tests/MongoDB.AspNetCode.OData.Tests/TestServer.cs

```diff
@@ -0,0 +1,113 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Json.Schema;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace MongoDB.AspNetCode.OData.Tests
+{
+    [TestClass]
+    internal static class TestServer
+    {
+        private static WebApplicationFactory<Program> _applicationFactory;
+
+        [AssemblyInitialize]
+        public static void TestServerInitialize(TestContext testContext)
+        {
+            _applicationFactory = new WebApplicationFactory<Program>();
+        }
+
+        [AssemblyCleanup]
+        public static void TestServerInitializeCleanup()
+        {
+            _applicationFactory.Dispose();
+        }
+
+        public static async Task UseHttpClientAsync(Func<HttpClient, Task> fn)
```

Comment:
minor: `fn`, better to use more descriptive naming.

Also consider having `Task<JsonDocument> GetAsync(string url)` instead of `UseHttpClientAsync`. Looks like `UseHttpClientAsync` is used only internally, so maybe it's worth exposing such general method only when needed.

Reply:
Done